### PR TITLE
server spawning tequila shots with shattering glasses

### PR DESCRIPTION
**Pull request by RebeccaStankus**

### CODE REVIEW

File: DomainContent/Mexico/tequilaServer.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// Author: Liv Erickson
+// Copyright High Fidelity 2018
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+//
+/* globals Entities */
+(function(){
+  
+    var PIECE_MODEL = "http://hifi-content.s3.amazonaws.com/rebecca/Hub/Space%20Juice/models/shot-glass-fragment.fbx";
```
Comment (misslivirose) [inline DomainContent/Mexico/tequilaServer.js]: Dev URL, should be production location or using script.resolve

File: DomainContent/Mexico/tequilaShotPrimitiveCatch.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    Tequila.prototype = {
+        preload: function(entityID) {
+            _this = this;
+            _this.entityID = entityID;
+            glass = Entities.getEntityProperties(_this.entityID, 'parentID').parentID;
+            spawner = Entities.getEntityProperties(glass, 'parentID').parentID;
+        },
+        mouseReleaseOnEntity: function(entityID, mouseEvent) {
+            print("calling tequila client");
```
Comment (misslivirose) [inline DomainContent/Mexico/tequilaShotPrimitiveCatch.js]: Remove print statements

File: DomainContent/Mexico/tequilaSpawnerServer.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    TequilaSpawner.prototype = {
+        remotelyCallable: ['spawnIfNeeded', 'spawnNewTequilaShot'],
+        preload: function(entityID) {
+            print("good morning sunshine");
+            _this.entityID = entityID;
+            _this.position = Entities.getEntityProperties(_this.entityID, 'position').position;
+            Script.setInterval(function(){
+                print("interval");
```
Comment (misslivirose) [inline DomainContent/Mexico/tequilaSpawnerServer.js]: Remove print statement, space between () and {

File: DomainContent/Mexico/tequilaSpawnerServer.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                if (name === "Tequila Shot") {
+                    needToSpawn = false;
+                }
+            });
+            if (needToSpawn) {
+                _this.spawnNewTequilaShot();
+            }
+        },
+
+        createDrunkZone: function() {
```
Comment (misslivirose) [inline DomainContent/Mexico/tequilaSpawnerServer.js]: Is this the same zone that's created in the client?

Comment (RebeccaStankus) [reply DomainContent/Mexico/tequilaSpawnerServer.js]: It's created in the server script, the client only checks to see if it's there in order to see if the user already has been drinking.

File: DomainContent/Mexico/tequilaClient.js [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                                lifetime: 30
+                            }});
+                            break;
+                        case 50:
+                            Entities.editEntity(drunkZone, {haze: {
+                                hazeRange: 10,
+                                hazeBackgroundBlend: 0.1,
+                                lifetime: 30
+                            }});
+                            _this.fallDown();
```
[the remark this answers is not in the document]
Comment (RebeccaStankus) [reply DomainContent/Mexico/tequilaClient.js]: if the user drinks 4 shots in a short enough interval of time, they fall down. If they continue to drink quickly enough, before the drunken haze has cleared, they fall down again.

File: DomainContent/Mexico/tequilaClient.js [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    var breakURL = "https://hifi-content.s3.amazonaws.com/liv/dev/250709__aiwha__glass-break-2.wav";
+    var breakSound = SoundCache.getSound(breakURL);
+    var volumeLevel = 0.65;
+    var _entityID;
+
+    function TequilaGlass() {
+        _this = this;
+    }
+
+    var shouldBreak = function(velocity){
```
[the remark this answers is not in the document]
Comment (misslivirose) [reply DomainContent/Mexico/tequilaClient.js]: Also space after ) before {

**Comment by misslivirose** [comment]
@RebeccaStankus Where are these staged for testing purposes?

**Comment by RebeccaStankus** [comment]
They were in Mexico, but it looks like they've been deleted. I'll get them back and let you know when they're ready.

**Comment by MiladNazeri** [comment]
@RebeccaStankus does this still need code review or can the label be removed?

**Comment by misslivirose** [comment]
This is an old PR but it's been internally tested and deployed, so marking and merging.

**Pull request merged**